### PR TITLE
ci: Implement code scanning with CodeQL

### PR DESCRIPTION
**Pull request by reedloden**
Use CodeQL to scan all pull requests and pushes for known security issues.

https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-with-codeql

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", branch/* ]
+  pull_request:
+    branches: [ "master", branch/* ]
+  schedule:
+    - cron: '26 9 * * 6'
```
Comment (zmb3) [inline .github/workflows/codeql-analysis.yml:9]: Is the cron schedule necessary if this is going to run on every push and PR?

**Review by zmb3 — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (reedloden) [reply .github/workflows/codeql-analysis.yml:9]: Probably not with our activity. A weekly job is provided by default.

https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#scanning-on-a-schedule

It's generally meant for projects that may not be committing all the time, just to make sure that CodeQL is running, as new queries and such pop up often.

I'll go ahead and remove it.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'javascript', 'python' ]
```
Comment (r0mant) [inline .github/workflows/codeql-analysis.yml:21]: Do we have any Python code?

**Review by r0mant — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (reedloden) [reply .github/workflows/codeql-analysis.yml:25]: Technically, we have:
* `./examples/aws/terraform/ha-autoscale-cluster/ansible/ec2.py`
* `./assets/monitoring/convert.py`

But to save build time, I'll remove it.

**Pull request merged**